### ISSUE TITLE
sqlite3 SELECT does not BEGIN a transaction, but should according to spec

### ISSUE THREAD

**Issue by 9a086caa-4eb7-43c8-8523-125e6985f4c3** [python/cpython#54133]
BPO | [9924](https://bugs.python.org/issue9924)
--- | :---
Nosy | @pitrou, @bitdancer, @dholth, @berkerpeksag, @palaviv, @erlend-aasland
Files | <li>[unintuitive_sqlite_behavior.py](https://bugs.python.org/file42260/unintuitive_sqlite_behavior.py "Uploaded as text/plain at 2016-03-23.16:38:51 by rhunter")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2010-09-23.03:56:05.034>
labels = ['3.7', 'type-bug', 'library']
title = 'sqlite3 SELECT does not BEGIN a transaction, but should according to spec'
updated_at = <Date 2021-05-19.11:16:15.521>
user = 'https://bugs.python.org/zzzeek'
```

bugs.python.org fields:
```python
activity = <Date 2021-05-19.11:16:15.521>
actor = 'erlendaasland'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2010-09-23.03:56:05.034>
creator = 'zzzeek'
dependencies = []
files = ['42260']
hgrepos = []
issue_num = 9924
keywords = []
message_count = 7.0
messages = ['117167', '117168', '117180', '129128', '207794', '262284', '307684']
nosy_count = 11.0
nosy_names = ['ghaering', 'pitrou', 'rnortman', 'r.david.murray', 'zzzeek', 'dholth', 'berker.peksag', 'palaviv', 'rhunter', 'Allen Li', 'erlendaasland']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue9924'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by erlend-aasland** [comment]
As of the recent [discussion on Discourse](https://discuss.python.org/t/introducing-a-pep-249-compliant-transaction-behaviour-to-sqlite3/16516?u=erlendaasland), I'm closing this as superseded by the proposed solution in gh-83638.

**TL;DR**: We cannot change the behaviour because of backwards compatibility. The proposed fix is to introducing a new, PEP 249 compliant way of controlling transactions. The old transaction handling (`isolation_level`) will be deprecated.
